mobile-app/app/screens/AppNavigator/screens/Dex/DexScreen.tsx

```diff
@@ -180,18 +184,21 @@ export function DexScreen (): JSX.Element {
   }, [])
 
   useEffect(() => {
-    setIsSearching(true)
-    handleFilter(searchString)
+    if (showSearchInput) {
+      setIsSearching(true)
+      handleFilter(searchString)
+    }
   }, [searchString, hasFetchedPoolpairData])
 
   // Update local state - filter available pair when pairs update
   useEffect(() => {
-    if (searchString !== undefined && searchString.trim().length > 0) {
-      handleFilter(searchString)
+    if (showSearchInput) {
```

Comment:
If possible we would try to avoid nested if-condition statement, this is mainly for readability 
e.g.: we can do an early return for `if (!showSearchInput) {... return}`

Reply:
Not sure whether that makes it more readable, but I will do that